udata/core/dataservices/rdf.py

```diff
@@ -71,11 +70,9 @@ def dataservice_from_rdf(
                 None,
             )
 
-        if dataset is not None:
-            datasets.append(dataset.id)
-
-    if datasets:
-        dataservice.datasets = datasets
+        # We append the dataset to the list of the current attached ones if not already attached
+        if dataset is not None and dataset not in dataservice.datasets:
```

Comment:
`dataset not in dataservice.datasets` is it working? Could you make sur we have a test on this because I'm always septical about comparing objects like this :) (we may need to compare IDs?)